File: SampleApp/Objective-C/SampleApp/AdMob/RewardedVideo/AdMobRewardedVideoViewController.m
//
//  AdMobRewardedVideoViewController.m
//  SampleApp
//
//  Copyright © 2020 fluct, inc. All rights reserved.
//

#import "AdMobRewardedVideoViewController.h"
#import <FluctSDK/FluctSDK.h>
#import <GoogleMobileAds/GoogleMobileAds.h>
#import <GoogleMobileAdsMediationFluct/GADMAdapterFluctExtras.h>

@interface AdMobRewardedVideoViewController () <GADFullScreenContentDelegate>
@property (nonatomic, weak) IBOutlet UIButton *showButton;
@property (nonatomic, nullable) GADRewardedAd *rewardedAd;
@end

// AdUnitIDを適切なものに変えてください
static NSString *const kAdUnitID = @"ca-app-pub-3010029359415397/4697035240";

@implementation AdMobRewardedVideoViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (IBAction)didTouchUpLoadAdWithButton:(UIButton *)button {
    GADRequest *request = [GADRequest request];

    FSSRewardedVideoSetting *setting = FSSRewardedVideoSetting.defaultSetting;
    setting.debugMode = YES;
    setting.activation.unityAdsActivated = NO;

    GADMAdapterFluctExtras *extras = [GADMAdapterFluctExtras new];
    extras.setting = setting;

    [request registerAdNetworkExtras:extras];

    [GADRewardedAd loadWithAdUnitID:kAdUnitID
                            request:request
                  completionHandler:^(GADRewardedAd *_Nullable rewardedAd, NSError *_Nullable error) {
                      if (error) {
                          NSLog(@"loadRequest:comptionHander error:   %@", error);
                          return;
                      }

                      self.rewardedAd = rewardedAd;
                      self.rewardedAd.fullScreenContentDelegate = self;
                      self.showButton.enabled = YES;
                  }];
}

- (IBAction)didTouchUpShowAdWithButton:(UIButton *)button {
    [self.rewardedAd presentFromRootViewController:self
                          userDidEarnRewardHandler:^{
                              NSLog(@"%s, %@", __FUNCTION__, self.rewardedAd.adReward);
                          }];
}

#pragma mark - GADFullScreenContentDelegate

- (void)adDidPresentFullScreenContent:(id<GADFullScreenPresentingAd>)ad {
    NSLog(@"%s", __FUNCTION__);
}

- (void)adDidDismissFullScreenContent:(id<GADFullScreenPresentingAd>)ad {
    NSLog(@"%s", __FUNCTION__);
    self.showButton.enabled = NO;
}

- (void)ad:(id<GADFullScreenPresentingAd>)ad didFailToPresentFullScreenContentWithError:(NSError *)error {
    NSLog(@"%s, %@", __FUNCTION__, error);
}

@end

File: SampleApp/Swift/SampleApp/AdMob/RewardedVideo/AdMobRewardedVideoViewController.swift
//
//  AdMobRewardedVideoViewController.swift
//  SampleApp
//
//  Copyright © 2020 fluct, Inc. All rights reserved.
//

import UIKit
import FluctSDK
import GoogleMobileAdsMediationFluct
import GoogleMobileAds

// AdUnitIDを適切なものに変えてください
private let adUnitID: String = "ca-app-pub-3010029359415397/4697035240"

class AdMobRewardedVideoViewController: UIViewController {

    @IBOutlet weak var showButton: UIButton!

    private var rewardedAd: GADRewardedAd?

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }

    @IBAction func didTouchUpLoadAd(button: UIButton) {
        let request = GADRequest()

        let setting = FSSRewardedVideoSetting.default
        // UnityAdsだけ再生しない
        setting.activation.isUnityAdsActivated = false
        setting.isDebugMode = true
        let extra = GADMAdapterFluctExtras()
        extra.setting = setting
        request.register(extra)

        GADRewardedAd.load(withAdUnitID: adUnitID, request: request) {[weak self] (ad, error) in
            if let error = error {
                print(error)
                return
            }
            self?.rewardedAd = ad
            self?.rewardedAd?.fullScreenContentDelegate = self
            self?.showButton.isEnabled = true
        }
    }

    @IBAction func didTouchUpShowAd(button: UIButton) {
        guard let rewardedAd = self.rewardedAd else { return }
        rewardedAd.present(fromRootViewController: self) {[rewardedAd] in
            print(rewardedAd.adReward)
        }
    }

}

extension AdMobRewardedVideoViewController: GADFullScreenContentDelegate {

    func ad(_ ad: GADFullScreenPresentingAd, didFailToPresentFullScreenContentWithError error: Error) {
        print(#function, error)
    }

    func adDidPresentFullScreenContent(_ ad: GADFullScreenPresentingAd) {
        print(#function)
    }

    func adDidRecordImpression(_ ad: GADFullScreenPresentingAd) {
        print(#function)
    }

    func adDidDismissFullScreenContent(_ ad: GADFullScreenPresentingAd) {
        print(#function)
        self.showButton.isEnabled = false
    }
}
